DM-13993: Update package creation documentation (#396)
lsst-dm/dm_dev_guide: stack/adding-a-new-package.rst

Before adding a new dependency to any of these products, it must be approved through the :doc:`RFC process </processes/decision_process>`.
Consensus must be reached regarding both the name and the suitability of the new package.

> Before adopting the RFC, The following steps must be completed:

Reviewer: ```suggestion
Before adopting the RFC, the following steps must be completed:
```

* Implementation tickets are created to cover package creation.
* The package is migrated to the ``lsst`` org, if not already there.

> * An audit of any dependencies with a focus on identifying implied dependencies must be completed.

Reviewer: ```suggestion
* An audit is done of any dependencies with a focus on identifying implied dependencies.
```
